server/src/infra/repositories/server-info.repository.ts

```diff
@@ -1,12 +1,13 @@
 import { GitHubRelease, IServerInfoRepository } from '@app/domain';
 import { Injectable } from '@nestjs/common';
-import axios from 'axios';
 
 @Injectable()
 export class ServerInfoRepository implements IServerInfoRepository {
-  getGitHubRelease(): Promise<GitHubRelease> {
-    return axios
-      .get<GitHubRelease>('https://api.github.com/repos/immich-app/immich/releases/latest')
-      .then((response) => response.data);
+  async getGitHubRelease(): Promise<GitHubRelease> {
+    const response = await fetch('https://api.github.com/repos/immich-app/immich/releases/latest');
+    if (!response.ok) {
+      throw new Error('Failed to fetch GitHub release');
```

Comment:
can you log or throw the actual details of the failure? otherwise it's extremely frustrating to troubleshoot